### ISSUE TITLE
NewType(Annotated..., ...) is not being flagged

### ISSUE THREAD

**Issue by FynnBe** [microsoft/pyright#7383]
"Hiding" an Annotated type behind a NewType does allow calling it:

```python
from typing import Annotated, NewType

A = Annotated[int, "whatever"]

B = NewType("B", A)

_ = B(5)

```

_Originally posted by @FynnBe in https://github.com/microsoft/pyright/issues/7370#issuecomment-1972050047_

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.353, which I just published.
